main/src/cgeo/geocaching/sorting/SortActionProvider.java

```diff
@@ -99,6 +100,9 @@ private void register(@StringRes final int resourceId, final Class<? extends Cac
 
     private void registerComparators() {
         registry.clear();
+        if (distanceToTargetAvailable) {
+            register(R.string.caches_sort_distance_target, TargetDistanceComparator.class, () -> TargetDistanceComparator.DISTANCE_TO_TARGET);
```

Comment:
Don't use ugly global constant here, pass target coords in a constructor to `TargetDistanceComparator` instead.

The reason why this is different for distance to user is explained in my second comment.